packages/react-components/src/components/chart/hooks/useHandleChartEvents.ts

```diff
@@ -0,0 +1,115 @@
+import { MutableRefObject, useEffect, useState } from 'react';
+import { EChartsOption, EChartsType } from 'echarts';
+
+export const useHandleChartEvents = (chartRef: MutableRefObject<EChartsType | null>) => {
+  const [optionState, setOptionState] = useState<EChartsOption>({});
+
+  // When dragging on the chart, turn off tooltip and set cursor to grabbing
+  const setOnDragStart = (chart: EChartsType | null) => {
+    setOptionState({
+      tooltip: {
+        show: false,
+      },
+    });
+    chart?.getZr().setCursorStyle('grabbing');
+  };
+
+  // When dragging is done turn back on the tooltip and set cursor to grab
+  // Use when the shift key is still held down but mouse is up
+  const setOnDragEnd = (chart: EChartsType | null) => {
+    setOptionState({
+      tooltip: {
+        show: true,
+      },
+    });
+    chart?.getZr().setCursorStyle('grab');
+  };
+
+  // Set up event listeners for pressing shift and dragging the mouse to update
+  // the cursor and tooltip
+  useEffect(() => {
+    const chart = chartRef?.current;
+
+    let shiftDown = false;
```

Comment:
curious: is this better than using local state ?

Reply:
Not using these variables for any re-renders, so only need to manage the logic within the `useEffect`